File: Daily/2023/1502_canMakeArithmeticProgressionFromSequenc.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        n = len(arr)
        arr.sort()
        diff = arr[1] - arr[0]
        for idx in range(2, n):
            if arr[idx] - arr[idx - 1] != diff:
                return False

        return True

File: Categories/Array/1502_canMakeArithmeticProgressionFromSequenc.cs
public class Solution {
    public bool CanMakeArithmeticProgression(int[] arr) {
        int n = arr.Length
        Array.Sort(arr)

        int diff = arr[1] - arr[0]

        for (int idx=2
             idx < n
             idx++) {
            if (arr[idx] - arr[idx - 1] != diff) {
                return false
            }
        }

        return true
    }
}
